src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NativeLayoutVertexNode.cs

```diff
@@ -256,12 +256,13 @@ public static NativeLayoutTypeSignatureVertexNode NewTypeSignatureVertexNode(Nod
                 case Internal.TypeSystem.TypeFlags.Class:
                 case Internal.TypeSystem.TypeFlags.ValueType:
                 case Internal.TypeSystem.TypeFlags.Interface:
+                case Internal.TypeSystem.TypeFlags.Nullable:
```

Comment:
It might be better to just replace all of this with a `default` case:
```csharp
default:
    Debug.Assert(type.IsDefType);
    ....
```

We already do the same in e.g. Transform.Type.cs.

Funny that I wanted to provide this as CR feedback yesterday, but I bit my tongue on that.